documentation/modules/cruise-control/con-optimization-goals.adoc

```diff
@@ -0,0 +1,133 @@
+
+Master optimization goals are controlled, if needed, in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.goals`
+
+* To accept the inherited master optimization goals, specify an empty value for the `goals` configuration option:
++
+[source=yaml]
+----
+goals: {}
+----
+
+* If you need to modify the inherited master optimization goals, specify a list of goals in the `goals` configuration option.
+
+[id="default-goals"]
+[discrete]
+=== Default optimization goals
+
+We recommend that you always set _default optimization goals_ for the Kafka cluster.
+You can include any or all of the master optimization goals in the default optimization goals.
+
+Default optimization goals are set in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.default.goals`
+
+The default optimization goals generate the _cached optimization proposal_ from the workload model (unless you set user-provided optimization goals in a `KafkaRebalance` custom resource). 
+This is updated to reflect the current state of the Kafka cluster every 15 minutes by default. 
+If no default optimization goals are specified, the cached proposal is generated using the master optimization goals.
+
+[id="user-provided-goals"]
+[discrete]
+=== User-provided optimization goals
+
+_User-provided optimization goals_ narrow down the default goals.
+You can set them, if required, in the `KafkaRebalance` custom resource for a particular optimization proposal: `KafkaRebalance.spec.goals`
+
+They are useful for generating an optimization proposal that addresses a particular scenario.
+For example, you might want to optimize leader replica distribution across the Kafka cluster without considering goals for disk capacity or disk utilization. 
+So, you create a `KafkaRebalance` custom resource containing a user-provided goal for leader replica distribution only.
+
+User-provided optimization goals must:
+
+* Include all configured hard goals, or an error will occur
+* Be a subset of the master optimization goals
+
+To exclude the configured hard goals from an optimization proposal, add the `skiphardgoalscheck: true` option to the `KafkaRebalance` custom resource.
+
+[id="hard-soft-goals"]
+[discrete]
+=== Hard goals and soft goals
+
+In Cruise Control, six of the master optimization goals are preset as _hard goals_:
+
+[source]
+RackAwareGoal; ReplicaCapacityGoal; DiskCapacityGoal; NetworkInboundCapacityGoal; NetworkOutboundCapacityGoal; CpuCapacityGoal
+
+All configured hard goals _must_ be satisfied in optimization proposals (unless `skip_hard_goal_check: true` is specified in the `KafkaRebalance` custom resource).
+In other words, an optimization proposal that does _not_ satisfy all the hard goals is rejected by Cruise Control and not sent to the user.
+
+Hard goals are controlled in the Cruise Control deployment configuration: `Kafka.spec.cruiseControl.config.hard.goals`
```

Comment:
the same as before, `hard.goals` is a property as a whole, we should clarify as I already mentioned for `default.goals`